### PR TITLE
[Feat] [snt9|2_5] Add a case: ShuffleNetV1-Based food recognition

### PR DESCRIPTION
**Pull request by ilovexsir**
## Test environment
Hardware: 1*ascend-snt9|ARM: 24核 96GB
Image: python3.9-ms2.5.0-cann8.0.0.beta1
## Test Report

Accuracy: 69.6%
Inference Time: 16.4 ms per image
Training Time: 10 minutes
Total Script Runtime: 16 minutes
## Related Issues

Fixes #4

### CODE REVIEW

**Comment by hongziqi** [comment]
It is recommended to consider optimizing the script to improve the model accuracy to at least 60% first.

**Comment by hongziqi** [comment]
I actually ran it with only 49% accuracy
image

File: dev/ascend/snt9/2_5/ShuffleNetV1实现食物识别.ipynb [hunk 1]
```diff
@@ -0,0 +1,712 @@
...
+    "ascend_target = False\n",
+    "if ascend_target:\n",
+    "    model = train.Model(network, loss_fn=network_loss, optimizer=network_opt, metrics={\"acc\"}, amp_level=\"O2\")\n",
+    "else:\n",
+    "    model = train.Model(network, loss_fn=network_loss, optimizer=network_opt, metrics={\"acc\"}, amp_level=\"O0\")\n",
+    "\n",
+    "# train model\n",
+    "model.train(epoch_size,\n",
+    "            dataset_train,\n",
+    "            callbacks=[ckpt_callback, LossMonitor(step_size), TimeMonitor(step_size)],\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/ShuffleNetV1实现食物识别.ipynb:538]:
```suggestion
    "            callbacks=[ckpt_callback, LossMonitor(step_size)],\n",
```
please remove TimeMonitor if epoch > 10

Comment (ilovexsir) [reply dev/ascend/snt9/2_5/ShuffleNetV1实现食物识别.ipynb]: Done

File: dev/ascend/snt9/2_5/ShuffleNetV1实现食物识别.ipynb [hunk 2]
```diff
@@ -0,0 +1,701 @@
...
+    "                                  smooth_factor=0.1,\n",
+    "                                  num_classes=num_classes)\n",
+    "\n",
+    "# set checkpoint\n",
+    "ckpt_config = CheckpointConfig(save_checkpoint_steps=step_size, keep_checkpoint_max=5)\n",
+    "ckpt_callback = ModelCheckpoint(prefix='InceptionResnetV2', directory='./InceptionResnetV2', config=ckpt_config)\n",
+    "\n",
+    "# initialize model\n",
+    "# \"Ascend + mixed precision\" can improve performance\n",
+    "#ascend_target = (ms.get_context(\"device_target\") == \"Ascend\")\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/ShuffleNetV1实现食物识别.ipynb:518]:
```suggestion
```

File: dev/ascend/snt9/2_5/ShuffleNetV1实现食物识别.ipynb [hunk 3]
```diff
@@ -0,0 +1,701 @@
...
+    "ascend_target = False\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/ShuffleNetV1实现食物识别.ipynb:519]:
```suggestion
```

File: dev/ascend/snt9/2_5/ShuffleNetV1实现食物识别.ipynb [hunk 4]
```diff
@@ -0,0 +1,701 @@
...
+    "if ascend_target:\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/ShuffleNetV1实现食物识别.ipynb:520]:
```suggestion
```

File: dev/ascend/snt9/2_5/ShuffleNetV1实现食物识别.ipynb [hunk 5]
```diff
@@ -0,0 +1,701 @@
...
+    "    model = train.Model(network, loss_fn=network_loss, optimizer=network_opt, metrics={\"acc\"}, amp_level=\"O2\")\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/ShuffleNetV1实现食物识别.ipynb:521]:
```suggestion
```

File: dev/ascend/snt9/2_5/ShuffleNetV1实现食物识别.ipynb [hunk 6]
```diff
@@ -0,0 +1,701 @@
...
+    "                                  reduction=\"mean\",\n",
+    "                                  smooth_factor=0.1,\n",
+    "                                  num_classes=num_classes)\n",
+    "\n",
+    "# set checkpoint\n",
+    "ckpt_config = CheckpointConfig(save_checkpoint_steps=step_size, keep_checkpoint_max=5)\n",
+    "ckpt_callback = ModelCheckpoint(prefix='InceptionResnetV2', directory='./InceptionResnetV2', config=ckpt_config)\n",
+    "\n",
+    "# initialize model\n",
+    "# \"Ascend + mixed precision\" can improve performance\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/ShuffleNetV1实现食物识别.ipynb:517]:
```suggestion
    "# Not using mixed precision\n",
```

File: dev/ascend/snt9/2_5/ShuffleNetV1实现食物识别.ipynb [hunk 7]
```diff
@@ -0,0 +1,701 @@
...
+    "#ascend_target = (ms.get_context(\"device_target\") == \"Ascend\")\n",
+    "ascend_target = False\n",
+    "if ascend_target:\n",
+    "    model = train.Model(network, loss_fn=network_loss, optimizer=network_opt, metrics={\"acc\"}, amp_level=\"O2\")\n",
+    "else:\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/ShuffleNetV1实现食物识别.ipynb:522]:
```suggestion
```

File: dev/ascend/snt9/2_5/ShuffleNetV1实现食物识别.ipynb [hunk 8]
```diff
@@ -0,0 +1,701 @@
...
+    "    model = train.Model(network, loss_fn=network_loss, optimizer=network_opt, metrics={\"acc\"}, amp_level=\"O0\")\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/ShuffleNetV1实现食物识别.ipynb:523]:
```suggestion
    "model = train.Model(network, loss_fn=network_loss, optimizer=network_opt, metrics={\"acc\"}, amp_level=\"O0\")\n",
```

**Review by hongziqi — APPROVED** [review]
